# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerControllerChild.cpp
# Commit: 48eaa84c206d
# Full Hash: 48eaa84c206dbdda637c57b55ddbd07e975ba103
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-06-23 16:15:06
# Regressor Bug: 1637500
# File Overlap Count: 2
# Description:
#   Bug 1637500 - P2. Rename methods as they are not always dealing with "threads". r=froydnj
#   
#   Before P1, GetCurrentThreadSerialEventTarget would have always returned the same data as NS_GetCurrentThread, making the comment incorrect Now it will properly return the running TaskQueue if any.
#   
#   This change of name more clearly exposes what they are doing, as we aren't always dealing with threads directly; but a nsISerialEventTarget
# ==============================================================================

diff -r 00687d06a8a6 -r 48eaa84c206d dom/workers/remoteworkers/RemoteWorkerControllerChild.cpp
--- a/dom/workers/remoteworkers/RemoteWorkerControllerChild.cpp	Tue Jun 23 05:02:39 2020 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerControllerChild.cpp	Tue Jun 23 05:05:36 2020 +0000
@@ -104,7 +104,7 @@
   if (mObserver) {
     static_cast<ServiceWorkerPrivateImpl*>(mObserver.get())
         ->SetSkipWaitingFlag()
-        ->Then(GetCurrentThreadSerialEventTarget(), __func__,
+        ->Then(GetCurrentSerialEventTarget(), __func__,
                [resolve = std::move(aResolve)](
                    const GenericPromise::ResolveOrRejectValue& aResult) {
                  resolve(aResult.IsResolve() ? aResult.ResolveValue() : false);
@@ -137,7 +137,7 @@
   RefPtr<RemoteWorkerControllerChild> self = this;
 
   SendShutdown()->Then(
-      GetCurrentThreadSerialEventTarget(), __func__,
+      GetCurrentSerialEventTarget(), __func__,
       [self = std::move(self)](const ShutdownPromise::ResolveOrRejectValue&) {
         if (self->mIPCActive) {
           Unused << self->Send__delete__(self);